src/storage/src/render/persist_sink.rs

```diff
@@ -123,20 +834,24 @@ where
         // to. Data from the source not beyond this time will be dropped, as it has already
         // been persisted.
         // In the future, sources will avoid passing through data not beyond this upper
-        if active_write_worker {
+        if active_worker {
```

Comment:
ditto here, can this be moved before the open_writer call?

Reply:
we need the `write.upper()` here

Reply:
oh wait on the false can I probably can!